sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -78,6 +78,27 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     }
   }
 
+  private def toSQL(node: LogicalPlan, topNode: Boolean): String = {
```

Comment:
Actually, I am also afraid that the test case here for `Join` is not valid. I guess `Join` is the top layer node only when users are using DataFrame/DataSet APIs. If users are using SQL, `Join` will not be the top layer node, right?

If we want to convert the analyzed logical plan generated from DataFrame API to SQL, this is not the only issue. I also hit a couple of issues before. @liancheng , do you think we should support them?